core/commands/root.go

```diff
@@ -19,6 +19,17 @@ var log = logging.Logger("core/commands")
 var ErrNotOnline = errors.New("this command must be run in online mode. Try running 'ipfs daemon' first")
 
 const (
+	// FIXME(BLOCKING): Revisit these names. They were chosen for the review
+	//  as they make explicit whether they signal a directory or a file (and
+	//  we needed to differentiate them from the old "config" option).
+	RepoDirOption    = "repo-dir"
+	ConfigFileOption = "config-file"
+	// FIXME(BLOCKING): Decide what to do when the user sets the deprecated
+	//  (and incorrectly named) "config" option, either:
+	//  * fail entirely
+	//  * issue a warning but process it as if it were the old equivalent RepoDirOption
+	//    (and fail if either of the new options that replace it are set)
```

Comment:
Do we have some precedence for how we've done this in the past? Abstractly I'd prefer this option but we should probably do the least surprising thing, whatever that is (if it's to just make the breaking change, then IMO we should just do it).

Reply:
I don't know; will need someone else to make that call.

Reply:
now that we are running with "deprecate", can we remove this comment?